scripts/check-monitoring.go

```diff
@@ -0,0 +1,181 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+//go:generate go run check-monitoring.go ../monkit.lock
+
+// +build ignore
+
+package main
+
+import (
+	"fmt"
+	"go/ast"
+	"go/token"
+	"go/types"
+	"io/ioutil"
+	"log"
+	"os"
+	"sort"
+	"strings"
+
+	"golang.org/x/tools/go/packages"
+)
+
+var (
+	monkitPath    = "gopkg.in/spacemonkeygo/monkit.v2"
+	lockFilePerms = os.FileMode(0644)
+)
+
+func main() {
+	pkgs, err := packages.Load(&packages.Config{
+		Mode: packages.NeedCompiledGoFiles | packages.NeedSyntax | packages.NeedName | packages.NeedTypes | packages.NeedTypesInfo,
+	}, "storj.io/storj/...")
+	if err != nil {
+		log.Fatalf("error while loading packages: %s", err)
+	}
+
+	var lockedFnNames []string
+	for _, pkg := range pkgs {
+		lockedFnNames = append(lockedFnNames, findLockedFnNames(pkg)...)
+	}
+	sortedNames := sortAndUnique(lockedFnNames)
+
+	outputStr := strings.Join(sortedNames, "\n")
+	if len(os.Args) == 2 {
+		file := os.Args[1]
+		if err := ioutil.WriteFile(file, []byte(outputStr+"\n"), lockFilePerms); err != nil {
+			log.Fatalf("error while writing to file \"%s\": %s", file, err)
+		}
+	} else {
+		fmt.Println(outputStr)
+	}
+}
+
+func findLockedFnNames(pkg *packages.Package) []string {
+	var (
+		lockedTasksPos []token.Pos
+		lockedTaskFns  []*ast.FuncDecl
+		lockedFnInfos  []string
+	)
+
+	// Collect locked comments and what line they are on.
+	for _, file := range pkg.Syntax {
+		lockedLines := make(map[int]struct{})
+		for _, group := range file.Comments {
+			for _, comment := range group.List {
+				if comment.Text == "//locked" {
+					commentLine := pkg.Fset.Position(comment.Pos()).Line
+					lockedLines[commentLine] = struct{}{}
+				}
+				if len(lockedLines) == 0 {
```

Comment:
I've just realized that this statement which I requested in my previous review has been placed in the wrong line https://github.com/storj/storj/pull/3114#discussion_r333424093

It has to be just after the scope of the `for _, group := range file.Comments` otherwise, it doesn't do anything.

Reply:
oops! thank you for catching!